### ISSUE TITLE
Deploying to Heroku

### ISSUE THREAD

**Issue by ChiefDakota** [ChiefDakota/react-on-rails-spa#13]
Have tried a few different things (a few too many times) and can't seem to get it to deploy to Heroku.  Either fails after compiling or in step 2/4 packaging dependencies. Maybe something wrong with my webpack config? 🤷‍♂️

**Comment by ChiefDakota** [comment]
Heroku doesn't seem to be recognizing the `Procfile`

**Comment by ChiefDakota** [comment]
without buildpacks:
```
remote:        Bundle completed (47.21s)
remote:        Cleaning up the bundler cache.
remote: -----> Installing node-v10.14.1-linux-x64
remote: -----> Installing yarn-v1.12.3
remote: -----> Detecting rake tasks
remote: -----> Preparing app for Rails asset pipeline
remote:        Running: rake assets:precompile
remote:        yarn install v1.12.3
remote:        [1/4] Resolving packages...
remote:        [2/4] Fetching packages...
remote:        error fsevents@1.2.7: The platform "linux" is incompatible with this module.
remote:        error Found incompatible module
remote:        info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
remote:        I, [2019-04-18T14:12:51.314277 #1439]  INFO -- : Writing /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/public/assets/application-13bf5de8a04c31ae70b41fb6c8feb4cedcac80a4025bdbc434e41c587df0a7e0.js
remote:        I, [2019-04-18T14:12:51.314672 #1439]  INFO -- : Writing /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/public/assets/application-13bf5de8a04c31ae70b41fb6c8feb4cedcac80a4025bdbc434e41c587df0a7e0.js.gz
remote:        I, [2019-04-18T14:12:51.320580 #1439]  INFO -- : Writing /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/public/assets/application-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css
remote:        I, [2019-04-18T14:12:51.320713 #1439]  INFO -- : Writing /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/public/assets/application-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css.gz
remote:        Webpacker is installed 🎉 🍰
remote:        Using /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/config/webpacker.yml file for setting up webpack paths
remote:        Compiling…
remote:        Compilation failed:
remote:        /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/vendor/bundle/ruby/2.5.0/gems/webpacker-3.6.0/lib/webpacker/webpack_runner.rb:11:in `exec': No such file or directory - /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/node_modules/.bin/webpack (Errno::ENOENT)
remote:        	from /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/vendor/bundle/ruby/2.5.0/gems/webpacker-3.6.0/lib/webpacker/webpack_runner.rb:11:in `block in run'
remote:        	from /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/vendor/bundle/ruby/2.5.0/gems/webpacker-3.6.0/lib/webpacker/webpack_runner.rb:10:in `chdir'
remote:        	from /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/vendor/bundle/ruby/2.5.0/gems/webpacker-3.6.0/lib/webpacker/webpack_runner.rb:10:in `run'
remote:        	from /tmp/build_ff5bdf38f773c7315e2116f5504a4bbb/vendor/bundle/ruby/2.5.0/gems/webpacker-3.6.0/lib/webpacker/runner.rb:6:in `run'
remote:        	from ./bin/webpack:15:in `<main>'
remote:
remote:
remote:
remote:  !
remote:  !     Precompiling assets failed.
remote:  !
remote:  !     Push rejected, failed to compile Ruby app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !	Push rejected to react-on-rails-spa.
remote:
To https://git.heroku.com/react-on-rails-spa.git
 ! [remote rejected] prod-deploy -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/react-on-rails-spa.git'
```

**Comment by ChiefDakota** [comment]
with Node > Ruby buildpacks:
```
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Node.js app detected
remote:
remote: -----> Creating runtime environment
remote:
remote:        NPM_CONFIG_LOGLEVEL=error
remote:        NODE_ENV=production
remote:        NODE_MODULES_CACHE=true
remote:        NODE_VERBOSE=false
remote:
remote: -----> Installing binaries
remote:        engines.node (package.json):  unspecified
remote:        engines.npm (package.json):   unspecified (use default)
remote:        engines.yarn (package.json):  unspecified (use default)
remote:
remote:        Resolving node version 10.x...
remote:        Downloading and installing node 10.15.3...
remote:        Using default npm version: 6.4.1
remote:        Resolving yarn version 1.x...
remote:        Downloading and installing yarn (1.15.2)...
remote:        Installed yarn 1.15.2
remote:
remote: -----> Installing dependencies
remote:        Installing node modules (yarn.lock)
remote:        yarn install v1.15.2
remote:        [1/4] Resolving packages...
remote:        [2/4] Fetching packages...
remote:        error fsevents@1.2.7: The platform "linux" is incompatible with this module.
remote:        error Found incompatible module
remote:        info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
remote:
remote: -----> Build failed
remote:
remote:        We're sorry this build is failing! You can troubleshoot common issues here:
remote:        https://devcenter.heroku.com/articles/troubleshooting-node-deploys
remote:
remote:        Some possible problems:
remote:
remote:        - Node version not specified in package.json
remote:          https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
remote:
remote:        Love,
remote:        Heroku
remote:
remote:  !     Push rejected, failed to compile Node.js app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !	Push rejected to react-on-rails-spa.
remote:
To https://git.heroku.com/react-on-rails-spa.git
 ! [remote rejected] prod-deploy -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/react-on-rails-spa.git'
```

**Comment by ChiefDakota** [comment]
Solved! Using the buildpacks, setting the versions, and/or removing `fsevents@1.2.7` fixed and successfully deployed! :)

**Comment by ChiefDakota** [comment]
Fixed with #14 and #15 😄
